gnat2goto/unit/floating_point_tests.adb

```diff
@@ -0,0 +1,142 @@
+with Test_Util;
+
+with Urealp; use Urealp;
+with Uintp; use Uintp;
+
+with Ureal_To_Binary; use Ureal_To_Binary;
+
+package body Floating_Point_Tests is
+
+   procedure Test_Integer;
+   procedure Test_Between_0_And_1;
+   procedure Test_General;
+
+   procedure Test_Negative_Integer;
+   procedure Test_Negative_Between_0_And_1;
+   procedure Test_Negative_General;
+
+   procedure Test_Zero;
+
+   procedure Test_Suite;
+
+   procedure Do_Test_Suite is
+   begin
+      Test_Util.Run_Test_Suite ("Floating point literals", Test_Suite'Access);
+   end Do_Test_Suite;
+
+   procedure Test_Integer is
+      --  16/2 = 8
+      Eight : constant Ureal := UR_From_Components (Uint_16, Uint_2);
```

Comment:
Just for clarification: So the type universal real is the type of the GNAT parsing result of some of the literals, but is not an actual ADA standard type?

Reply:
Yes, this is a GNAT internal type used to represent literals and constant expressions.